From 0a4f3aabc88105f92acfdda146dfeae25d4e134f Mon Sep 17 00:00:00 2001
From: Jeffy Chen <jeffy.chen@rock-chips.com>
Date: Wed, 19 Jul 2023 15:49:56 +0800
Subject: [PATCH 05/86] output-capture: Add a sanity check for source info when
 pulling task

Fix error when atomic_modeset disabled:
../libweston/output-capture.c:399: weston_output_pull_capture_task: Assertion `csi->width == width' failed.

Signed-off-by: Jeffy Chen <jeffy.chen@rock-chips.com>
---
 libweston/output-capture.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/libweston/output-capture.c b/libweston/output-capture.c
index 3265f06..2a6f4de 100644
--- a/libweston/output-capture.c
+++ b/libweston/output-capture.c
@@ -395,6 +395,10 @@ weston_output_pull_capture_task(struct weston_output *output,
 	 * already sent.
 	 */
 	csi = capture_info_get_csi(ci, src);
+
+	if (!source_info_is_available(csi))
+		return NULL;
+
 	assert(csi->width == width);
 	assert(csi->height == height);
 	assert(csi->drm_format == format->format);
-- 
2.20.1

